### ISSUE TITLE
Convert Address Input to a Single Field with Autocomplete

### ISSUE THREAD

**Issue by trevorskaggs** [trevorskaggs/shelterly#71]
Still want to save the full address details on the backend, but might be nice to integrate a nice Autocomplete on a single "Location" field on the front end.  Similiar to how Google does it, type a bit of the address and it tries to help you.

![image](https://user-images.githubusercontent.com/1276123/90946453-c629db00-e3e1-11ea-9589-472f01c889e5.png)

On the backend, you then can take the parts of the whole and parse them into the correct field.

**Comment by AlexMountain** [comment]
Did some looking into this.  Google seems like the defacto but is actually kind of expensive.  I think we can do better in the spirit of this project by at least trying an open source option pelias: https://github.com/pelias/pelias.  There are some handy autocomplete libraries already available for using this service:  https://github.com/generalpiston/react-autosuggest-geocoder.

I'm going to play with a temporary API key from geocode: https://geocode.earth/cloud, but it looks like we could potentially host our own server with docker: https://github.com/pelias/docker

**Comment by AlexMountain** [comment]
Hm, well, after some testing, unfortunately all of the Pelias examples I've found seem to be pretty outdated.  I installed `react-autosuggest-geocoder` but it's throwing an error and after messing with babel config for an hour I accepted defeat, at least for the time being.

error for reference:
```SyntaxError: /home/shelterly/frontend/node_modules/react-autosuggest-geocoder/src/geocoder.js: Support for the experimental syntax 'classProperties' isn't currently enabled (8:20):

Add @babel/plugin-proposal-class-properties (https://git.io/vb4SL) to the 'plugins' section of your Babel config to enable transformation.```

I dunno if someone else wants to take a stab at getting this thing to compile?

Setting up Google Places looks pretty quick and has modern examples. So depending on how much more time we want to spend vs expected costs for the services.
